### DISCUSSION TITLE
Аналог Префильтра во временной таблице

### DISCUSSION THREAD

**Discussion by tvideo** [totumonline/totum-ru-issues-and-discussions#96] [Неопределенное]
Здравствуйте. 
Может посоветуете мне с одним вопросиком в totum:
Есть временная таблица, которая формируется из входных данных. Первых два столбца селект-дерево, третий столбец - селект. Как сделать поиск по этим трем полям?
То есть что бы при выборе параметров поиска, строчная часть вывела только то, что подходит под поиск.
То есть замена Префильтра, так как его нельзя использовать во временных таблицах.
Спасибо
![1](https://user-images.githubusercontent.com/32614108/161394127-53efbc1a-638c-433a-b1fb-6ce0052d4a3c.png)
.

**Comment by mihail-cornev** [comment]
Нашли решение?

**Comment by poluninatm** [comment]
Мне видится такой вариант:

1. Входные данные строчной части временной таблицы складывать в поле в хедере (например, h_rows_data)
2. Строчную часть временной таблицы формировать через insert из h_insert, который будет себе копировать данные из h_rows_data
3. При первом расчете таблицы дополнить данные в h_rows_data вычисленными значениями дерева и селекта
4. В хедере сделать поле для фильтрации h_filter
5. Добавить в h_insert фильтр - если h_filter не пуст - то выводить только содержащие указанную строку строки.
